packages/markup-frontstage/src/services/SavedView.ts

```diff
@@ -147,22 +219,125 @@ export const createSavedViewData = (vp: Viewport): SavedViewData => {
 
   const data: SavedViewData = {
     alwaysDrawn,
-    neverDrawn,
     categories,
+    changeSetId,
+    displayStyleProps: JSON.stringify(viewState.displayStyle.toJSON()),
+    iModelId,
+    is2d: false,
     models,
-    version,
+    neverDrawn,
+    perModelCategoryVisibility,
     projectId,
-    iModelId,
-    changeSetId,
     sourceId,
-    userView: true,
     state,
-    displayStyleProps: JSON.stringify(viewState.displayStyle.toJSON()),
-    perModelCategoryVisibility,
+    userView: true,
+    version,
   };
 
   // Parse all data from emphasize elements (e.g. colorization, isolates, hidden, emphasize)
   parseFromEmphasizeElements(vp, data);
 
   return data;
 };
+
+/**
+ * Determines whether or not given saved view is spatial.
+ * @param view name of the Saved View.
+ */
+export const isSpatialSavedView = (view: SavedViewData) => {
+  return (view.is2d === undefined || !view.is2d) && "models" in view;
+};
+
+/**
+ * Determines whether or not given view is Drawing.
+ * @param view name fo the Saved View.
+ */
+export const isDrawingSavedView = (view: SavedViewData) => {
+  return view.is2d !== undefined && view.is2d && !("sheetProps" in view);
+};
+
+/**
+ * Determines whether or not given view is Sheet.
+ * @param view name of the Saved View.
+ */
+export const isSheetSavedView = (view: SavedViewData) => {
+  return view.is2d !== undefined && view.is2d && "sheetProps" in view;
+};
+
+/**
+ * Creates a drawing view state from the data object.
+ * @param iModelConnection the name of the IModel Connection.
+ * @param savedView the naem of the Saved View.
+ */
+export const createDrawingViewState = async (
```

Comment:
Similar comment as above. Could these be combined into a single parameterized function?

Reply:
combined both function to parameterize create2dViewState!